sdk/search/Azure.Search.Documents/src/Utilities/AggregateDictionary.cs

```diff
@@ -0,0 +1,311 @@
+        public bool IsReadOnly =>
+            _dictionaries.All(d => d.IsReadOnly);
+
+        /// <inheritdoc />
+        public void Add(TKey key, TValue value)
+        {
+            if (key is null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            if (ContainsKey(key))
+            {
+                throw new ArgumentException($"An element with the key '{key}' already exists", nameof(key));
+            }
+
+            Set(key, value);
+        }
+
+        /// <inheritdoc />
+        public void Add(KeyValuePair<TKey, TValue> item) =>
+            Add(item.Key, item.Value);
+
+        /// <inheritdoc />
+        public void Clear()
+        {
+            foreach (IDictionary<TKey, TValue> dictionary in _dictionaries)
+            {
+                dictionary.Clear();
+            }
+        }
+
+        /// <inheritdoc />
+        public bool Contains(KeyValuePair<TKey, TValue> item) =>
+            TryGetValue(item.Key, out TValue value) && EqualityComparer<TValue>.Default.Equals(value, item.Value);
+
+        /// <inheritdoc />
+        public bool ContainsKey(TKey key)
+        {
+            if (key is null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            foreach (IDictionary<TKey, TValue> dictionary in _dictionaries)
+            {
+                if (dictionary.ContainsKey(key))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <inheritdoc />
+        public void CopyTo(KeyValuePair<TKey, TValue>[] array, int arrayIndex)
+        {
+            if (array is null)
+            {
+                throw new ArgumentNullException(nameof(array));
+            }
+
+            foreach (IDictionary<TKey, TValue> dictionary in _dictionaries)
+            {
+                dictionary.CopyTo(array, arrayIndex);
+                arrayIndex += dictionary.Count;
+            }
+        }
+
+        /// <inheritdoc />
+        public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+        {
+            foreach (IDictionary<TKey, TValue> dictionary in _dictionaries)
+            {
+                foreach (KeyValuePair<TKey, TValue> pair in dictionary)
+                {
+                    yield return pair;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public bool Remove(TKey key)
+        {
+            if (key is null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            foreach (IDictionary<TKey, TValue> dictionary in _dictionaries)
+            {
+                if (dictionary.Remove(key))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <inheritdoc />
+        public bool Remove(KeyValuePair<TKey, TValue> item)
```

Comment:
That wouldn't be consistent with how the old dictionary works. `Remove(string key)` doesn't check the value. This variant of `Remove` has to, and does in the stock implementations in the .NET BCL. As mentioned elsewhere, all this is to make sure the existing `Configuration` property works exactly as it did before. We want to avoid any API or behavioral breaking changes.